app/js/actions/connection.js

```diff
@@ -48,12 +45,13 @@ const shakeHandsWithServer = () => {
     return (dispatch, getState) => {
         dispatch(updateConnectionStatus('Validating credentials'));
         const { connection: { uri, availableApis, secretKey } } = getState();
+        const url = `http://${uri.split('/')[0]}${availableApis[0]}`;
         const httpVerb = 'POST';
         const requestTime = Date.now();
         const body = JSON.stringify({});
-        const digest = makeB64Digest(secretKey, httpVerb, requestTime, body);
+        const digest = makeB64Digest(secretKey, httpVerb, url, requestTime, body);
```

Comment:
This can have the body variable deleted and `undefined` placed in the body position of `makeB64Digest()` call, as it's just sending the same thing that's in the default (catching my own mistakes here, haha).